### PR TITLE
Update the documentation of int_to_string() and to_string().

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/base/utilities.h [hunk 1]
```diff
@@ -177,10 +177,15 @@ namespace Utilities
...
-   * C function <code>itoa()</code> had been called.
+   * C++ `std::to_string` (or the older C function `itoa()`) had been called.
    *
-   * When calling this function signed integers are implicitly converted to
-   * unsigned integers and long integers might experience an overflow.
+   * This function takes an `unsigned int` as argument. As a consequence,
+   * if you call it with a `signed int` (which is of course the same
+   * type as `int`), the argument is implicitly converted to
+   * unsigned integers and negative numbers may not be printed as you had
+   * hoped. Similarly, if you cann the function with a `long int`, the
```
Comment (masterleinad) [inline include/deal.II/base/utilities.h]:
```suggestion
   * hoped. Similarly, if you call the function with a `long int`, the
```
?

**Comment by bangerth** [comment]
Oh yes, of course. Fixed.

File: include/deal.II/base/utilities.h [hunk 2]
```diff
@@ -177,10 +177,15 @@ namespace Utilities
...
    *
-   * When calling this function signed integers are implicitly converted to
-   * unsigned integers and long integers might experience an overflow.
+   * This function takes an `unsigned int` as argument. As a consequence,
+   * if you call it with a `signed int` (which is of course the same
+   * type as `int`), the argument is implicitly converted to
+   * unsigned integers and negative numbers may not be printed as you had
+   * hoped. Similarly, if you call the function with a `long int`, the
+   * printed result might show the effects of an overflow upon conversion
+   * to `unsigned int`.
```
Comment (tjhei) [inline include/deal.II/base/utilities.h]: can you suggest using to_string instead?

Comment (bangerth) [reply include/deal.II/base/utilities.h]: We already do three lines down :-)

**Comment by bangerth** [comment]
ping?

**Pull request merged**